api/v1alpha1/rabbitmqsecretengineconfig_types.go

```diff
@@ -0,0 +1,237 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"errors"
+	"reflect"
+
+	"github.com/redhat-cop/operator-utils/pkg/util/apis"
+	vaultutils "github.com/redhat-cop/vault-config-operator/api/v1alpha1/utils"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// RabbitMQSecretEngineConfigSpec defines the desired state of RabbitMQSecretEngineConfig
+type RabbitMQSecretEngineConfigSpec struct {
+	// Authentication is the k8s auth configuration to be used to execute this request
+	// +kubebuilder:validation:Required
+	Authentication KubeAuthConfiguration `json:"authentication,omitempty"`
+
+	// Path at which to make the configuration.
+	// The final path will be {[spec.authentication.namespace]}/{spec.path}/config/{metadata.name}.
+	// The authentication role must have the following capabilities = [ "create", "read", "update", "delete"] on that path.
+	// +kubebuilder:validation:Required
+	Path Path `json:"path,omitempty"`
+
+	// +kubebuilder:validation:Required
+	RMQSEConfig `json:",inline"`
+
+	// RootCredentials specifies how to retrieve the credentials for this RabbitMQEngine connection.
+	// +kubebuilder:validation:Required
+	RootCredentials RootCredentialConfig `json:"rootCredentials,omitempty"`
+}
+
+type RMQSEConfig struct {
+	// ConnectionURL Specifies the connection string used to connect to the RabbitMQ cluster.
+	// +kubebuilder:validation:Required
```

Comment:
this is a good place to add a regexp validation if there is a good one for a rabbitmq connection string.

Reply:
Sure, added simple regex  ^(http|https):\/\/.+$ for this, don't think anything more complex needed here

Reply:
this is good. I am concerned that in the case of a https connection one cannot pass the CA. it seems like a gap in the feature of this secret engine.

Reply:
Not sure which CA you would like to pass. If you are talking about connection between Vault Engine and RabbitMQ cluster there is no way to pass CA cert, it's expected for RabbitMQ to be configured with proper CA cert.

Reply:
ok